cylc/flow/scheduler.py

```diff
@@ -161,7 +160,6 @@ class SchedulerError(CylcError):
     """Scheduler expected error stop."""
 
 
-@dataclass
```

Comment:
Why remove this?

Reply:
This was causing incompatibility with Python 3.11 because some of the attributes are mutable which dataclass now rejects outright.

There wasn't an especially good reason for using dataclass. The main reason I put it in was to more explicitly define the scheduler's attributes (I think before type checking syntax came along, can't remember?).